pkg/kudoctl/cmd/package_list_plans.go

```diff
@@ -0,0 +1,129 @@
+  kubectl kudo package list plans zookeeper`
+)
+
+func newPackageListPlansCmd(fs afero.Fs, out io.Writer) *cobra.Command {
+	list := &packageListPlansCmd{fs: fs, out: out}
+
+	cmd := &cobra.Command{
+		Use:     "plans [operator]",
+		Short:   "List operator plans",
+		Example: packageListPlansExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := validateOperatorArg(args); err != nil {
+				return err
+			}
+			list.pathOrName = args[0]
+			return list.run(&Settings)
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVar(&list.RepoName, "repo", "", "Name of repository configuration to use. (default defined by context)")
+	f.StringVar(&list.PackageVersion, "version", "", "A specific package version on the official GitHub repo. (default to the most recent)")
+	f.BoolVarP(&list.WithTasksResources, "with-tasks", "t", false, "Display task resources with plans")
+
+	return cmd
+}
+
+func (c *packageListPlansCmd) run(settings *env.Settings) error {
+	pf, err := packageDiscovery(c.fs, settings, c.RepoName, c.pathOrName, c.PackageVersion)
+	if err != nil {
+		return err
+	}
+
+	return displayPlanTable(pf.Files, c.WithTasksResources, c.out)
+}
+
+func displayPlanTable(pf *packages.Files, withTasks bool, out io.Writer) error {
+	tree := treeprint.New()
+	planNames := sortedPlanNames(pf)
+	tree.SetValue("plans")
+	for _, name := range planNames {
+		plan := pf.Operator.Plans[name]
+		pNode := tree.AddBranch(fmt.Sprintf("%s (%s)", name, plan.Strategy))
+
+		for _, phase := range plan.Phases {
+			phNode := pNode.AddMetaBranch("phase", fmt.Sprintf("%s (%s)", phase.Name, phase.Strategy))
+			for _, step := range phase.Steps {
+				sNode := phNode.AddMetaBranch("step", step.Name)
+				for _, taskName := range step.Tasks {
+					if withTasks {
+						addTaskNodeWithResources(sNode, taskName, pf)
+					} else {
+						sNode.AddNode(taskName)
+					}
+				}
+			}
+		}
+	}
+
+	var err error
+	if len(pf.Operator.Plans) == 0 {
+		_, err = fmt.Fprintf(out, "no plans found\n")
+	} else {
+		_, err = fmt.Fprintln(out, tree.String())
+	}
+
+	return err
+}
+
+func sortedPlanNames(pf *packages.Files) []string {
+	planNames := make([]string, 0, len(pf.Operator.Plans))
+	for name := range pf.Operator.Plans {
+		planNames = append(planNames, name)
+	}
+	sort.Strings(planNames)
+	return planNames
+}
+
+func addTaskNodeWithResources(sNode treeprint.Tree, taskName string, pf *packages.Files) {
+	for _, t := range pf.Operator.Tasks {
+		if t.Name == taskName {
+			switch t.Kind {
+			case task.ApplyTaskKind:
+				tNode := sNode.AddMetaBranch("apply", taskName)
+				for _, resource := range t.Spec.Resources {
+					tNode.AddNode(resource)
+				}
+			case task.DeleteTaskKind:
+				tNode := sNode.AddMetaBranch("delete", taskName)
+				for _, resource := range t.Spec.Resources {
+					tNode.AddNode(resource)
+				}
+			case task.PipeTaskKind:
+				tNode := sNode.AddMetaBranch("pipe", taskName)
+				tNode.AddNode(t.Spec.Pod)
+			}
```

Comment:
please add `task.DummyTaskKind` and a `default`, so we still print new task kinds (with some empty/dummy) name.

Reply:
I'm not sure how I feel about this one... dummies shouldn't be here but it doesn't seem like it hurts.